### ISSUE TITLE
sound not working / youtube playing too fast / VLC video frozen

### ISSUE THREAD

**Issue by adrelanos** [QubesOS/qubes-issues#3685]
### Qubes OS version:
R3.2

### Affected component(s): 
debian, whonix
(fedora untested)

### Steps to reproduce the behavior:
go to youtube.com and watch some random video.

### Expected behavior:
youtube working normally with sound.

### Actual behavior:
* youtube video runs very fast, almost at double the speed.
* No sound.

### General notes:
* same behavior in both, Tor Browser and Firefox

```
dpkg -l | grep pulseaudio
ii  gstreamer1.0-pulseaudio:amd64                  1.10.4-1                                   amd64        GStreamer plugin for PulseAudio
ii  pulseaudio                                     10.0-1+deb9u1                              amd64        PulseAudio sound server
ii  pulseaudio-module-gconf                        10.0-1+deb9u1                              amd64        GConf module for PulseAudio sound server
ii  pulseaudio-utils                               10.0-1+deb9u1                              amd64        Command line tools for the PulseAudio sound server
```

```
sudo journalctl | grep pulse
Mar 09 12:57:10 host pulseaudio[1046]: vchan module loading
Mar 09 12:57:10 host pulseaudio[1046]: play libvchan_fd_for_select=19, ctrl=0x5f0ef5b93460
Mar 09 12:57:10 host pulseaudio[1046]: rec libvchan_fd_for_select=22, ctrl=0x5f0ef5b93ef0
Mar 09 12:57:10 host pulseaudio[1046]: sink cork req state =1, now state=-2
Mar 09 12:57:10 host pulseaudio[1046]: source cork req state =1, now state=-2
Mar 09 12:57:15 host pulseaudio[1046]: source cork req state =2, now state=1
Mar 09 12:57:15 host pulseaudio[1046]: sink cork req state =2, now state=1
Mar 09 12:59:23 host pulseaudio[1046]: sink cork req state =1, now state=2
Mar 09 12:59:23 host pulseaudio[1046]: sink cork req state =0, now state=1
Mar 09 12:59:27 host pulseaudio[1046]: sink cork req state =1, now state=0
Mar 09 12:59:27 host pulseaudio[1046]: sink cork req state =0, now state=1
Mar 09 12:59:46 host pulseaudio[1046]: sink cork req state =1, now state=0
Mar 09 12:59:48 host pulseaudio[1046]: write_to_vchan: all=8000 waited=289 nonwaited=7711 full=0
Mar 09 12:59:52 host pulseaudio[1046]: sink cork req state =2, now state=1
Mar 09 13:00:54 host pulseaudio[1046]: sink cork req state =1, now state=2
Mar 09 13:00:54 host pulseaudio[1046]: sink cork req state =0, now state=1
Mar 09 13:00:55 host pulseaudio[1046]: sink cork req state =1, now state=0
Mar 09 13:00:55 host pulseaudio[1046]: sink cork req state =0, now state=1
Mar 09 13:01:20 host pulseaudio[1046]: write_to_vchan: all=16000 waited=631 nonwaited=15369 full=0
Mar 09 13:01:49 host pulseaudio[1046]: write_to_vchan: all=24000 waited=1036 nonwaited=22964 full=0
Mar 09 13:02:17 host pulseaudio[1046]: write_to_vchan: all=32000 waited=1430 nonwaited=30570 full=0
Mar 09 13:02:45 host pulseaudio[1046]: write_to_vchan: all=40000 waited=1725 nonwaited=38275 full=0
Mar 09 13:03:13 host pulseaudio[1046]: write_to_vchan: all=48000 waited=2047 nonwaited=45953 full=0
Mar 09 13:03:42 host pulseaudio[1046]: write_to_vchan: all=56000 waited=2427 nonwaited=53573 full=0
Mar 09 13:04:10 host pulseaudio[1046]: write_to_vchan: all=64000 waited=2782 nonwaited=61218 full=0
Mar 09 13:04:25 host pulseaudio[1046]: sink cork req state =1, now state=0
Mar 09 13:04:25 host pulseaudio[1046]: sink cork req state =0, now state=1
Mar 09 13:04:37 host pulseaudio[1046]: sink cork req state =1, now state=0
Mar 09 13:04:37 host pulseaudio[1046]: sink cork req state =0, now state=1
Mar 09 13:04:37 host pulseaudio[1046]: sink cork req state =1, now state=0
Mar 09 13:04:40 host pulseaudio[1046]: write_to_vchan: all=72000 waited=3118 nonwaited=68882 full=0
Mar 09 13:04:42 host pulseaudio[1046]: sink cork req state =0, now state=1
Mar 09 13:04:53 host pulseaudio[1046]: sink cork req state =1, now state=0
Mar 09 13:04:58 host pulseaudio[1046]: sink cork req state =2, now state=1
```

* single notebook monitor (no dual monitor, not using HDMI)
* youtube and sound runs fine in Debian stretch based AppVMs

* VLC does not work either. No sound. Video frozen at first frame.

```
[00005c1bcd5f2148] core libvlc: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.
Failed to open VDPAU backend libvdpau_nvidia.so: cannot open shared object file: No such file or directory
Failed to open VDPAU backend libvdpau_nvidia.so: cannot open shared object file: No such file or directory
[00007dd4a4001268] xcb_xv vout display error: no available XVideo adaptor
Failed to open VDPAU backend libvdpau_nvidia.so: cannot open shared object file: No such file or directory
Failed to open VDPAU backend libvdpau_nvidia.so: cannot open shared object file: No such file or directory
Failed to open VDPAU backend libvdpau_nvidia.so: cannot open shared object file: No such file or directory
QObject::~QObject: Timers cannot be stopped from another thread
```

## VLC in Debian stretch based AppVM

for comparison 

```
VLC media player 2.2.7 Umbrella (revision 2.2.7-0-g6e32381286)
[00005cbc9654d0e8] core libvlc: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.
Failed to open VDPAU backend libvdpau_nvidia.so: cannot open shared object file: No such file or directory
Failed to open VDPAU backend libvdpau_nvidia.so: cannot open shared object file: No such file or directory
[00007e7f08001268] xcb_xv vout display error: no available XVideo adaptor
Failed to open VDPAU backend libvdpau_nvidia.so: cannot open shared object file: No such file or directory
Failed to open VDPAU backend libvdpau_nvidia.so: cannot open shared object file: No such file or directory
Failed to open VDPAU backend libvdpau_nvidia.so: cannot open shared object file: No such file or directory
Failed to open VDPAU backend libvdpau_nvidia.so: cannot open shared object file: No such file or directory
QObject::~QObject: Timers cannot be stopped from another thread
```

**Comment by adrelanos** [comment]
Mar 09 13:09:46 host qubes-gui[768]: ERROR reading WM_HINTS

This might be the cause?

```
sudo journalctl -u qubes-gui-agent
-- Logs begin at Fri 2018-03-09 12:57:08 UTC, end at Fri 2018-03-09 13:27:54 UTC. --
Mar 09 12:57:08 host systemd[1]: Starting Qubes GUI Agent...
Mar 09 12:57:08 host systemd[1]: Started Qubes GUI Agent.
Mar 09 12:57:08 host qubes-gui[768]: Waiting on /var/run/xf86-qubes-socket socket...
Mar 09 12:57:08 host su[806]: Successful su for user by root
Mar 09 12:57:08 host su[806]: + ??? root:user
Mar 09 12:57:08 host su[806]: pam_unix(su:session): session opened for user user by (uid=0)
Mar 09 12:57:09 host qubes-gui[768]: Ok, somebody connected.
Mar 09 12:58:26 host qubes-gui[768]: XGetWindowAttributes for 0xe00003 failed in handle_create, ret=0x0
Mar 09 12:58:26 host qubes-gui[768]: XGetWindowAttributes for 0xe0001c failed in handle_create, ret=0x0
Mar 09 12:58:39 host qubes-gui[768]: XGetWindowAttributes for 0x1200003 failed in handle_create, ret=0x0
Mar 09 12:58:39 host qubes-gui[768]: XGetWindowAttributes for 0x120001c failed in handle_create, ret=0x0
Mar 09 13:00:12 host qubes-gui[768]: XGetWindowAttributes for 0x1000003 failed in handle_create, ret=0x0
Mar 09 13:00:12 host qubes-gui[768]: XGetWindowAttributes for 0x100001c failed in handle_create, ret=0x0
Mar 09 13:00:23 host qubes-gui[768]: XGetWindowAttributes for 0x1400003 failed in handle_create, ret=0x0
Mar 09 13:00:23 host qubes-gui[768]: XGetWindowAttributes for 0x140001c failed in handle_create, ret=0x0
Mar 09 13:02:40 host qubes-gui[768]: XGetWindowAttributes for 0x1800018 failed in handle_create, ret=0x0
Mar 09 13:09:46 host qubes-gui[768]: ERROR reading WM_HINTS
Mar 09 13:10:15 host qubes-gui[768]: XGetWindowAttributes for 0x1800035 failed in handle_create, ret=0x0
Mar 09 13:12:07 host qubes-gui[768]: ERROR reading WM_HINTS
Mar 09 13:12:08 host qubes-gui[768]: CREATE for already existing 0x1800010
Mar 09 13:12:30 host qubes-gui[768]: XGetWindowAttributes for 0x1800264 failed in handle_create, ret=0x0
Mar 09 13:27:39 host qubes-gui[768]: XGetWindowAttributes for 0x1a00035 failed in handle_create, ret=0x0
```

**Comment by marmarek** [comment]
> This might be the cause?

Nope, gui should be totally separate.

Does Whonix have some audio related settings? I'd assume there is some additional isolation to prevent fingerprinting by listening on microphone or such (not really applicable in Qubes, but in non-Qubes probably very needed).
Try pavucontrol inside VM and see what output device is chosen (should be vchan something).

**Comment by adrelanos** [comment]
Marek Marczykowski-Górecki:
> Does Whonix have some audio related settings?

No.

> I'd assume there is some additional isolation to prevent fingerprinting by listening on microphone or such (not really applicable in Qubes, but in non-Qubes probably very needed).

We don't have any of that.

> Try pavucontrol inside VM and see what output device is chosen (should be vchan something).

Set to Qubes VCHAN sink. No difference Whonix vs Stretch.

anon-whonix:

pavucontrol

** (pavucontrol:2646): WARNING **: Error retrieving accessibility bus
address: org.freedesktop.DBus.Error.ServiceUnknown: The name
org.a11y.Bus was not provided by any .service files

stretch:

doesn't have that warning.

**Comment by adrelanos** [comment]
Found a workaround for me.

dom0 -> pavcontrol ->

* 1) set from `built in digital stereo (HDMI 2)` to
* 2) `set from built in audio analog stereo`.

Always reproducible. When a video is running, I can set back and forth to both settings:

* 1) always video running too fast without sound
* 2) video running normally with sound

Previously I had HDMI connected. Not now. Perhaps this is why I never run into this condition before.

-----

I just noticed this issue is not Whonix specific at all. It also happens in Debian. Fedora untested. Can test if that helps. Please kindly remove the Whonix tag. @andrewdavidwong 

-----

The bug here is:

dom0 sound setting shouldn't be set to `built in digital stereo (HDMI 2)` if HDMI gets disconnected.

**Comment by andrewdavidwong** [comment]
This issue is being closed because:

 - This issue has the "Release 3.2 updates" milestone.
 - [Qubes OS 3.2 recently reached end-of-life (EOL).](https://www.qubes-os.org/news/2019/03/28/qubes-3-2-has-reached-eol/)
 - This issue has not been updated in over a year.

If anyone believes that this issue should be reopened, please let us know in a comment here.
